src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -108,7 +101,7 @@ export const searchSourceRequiredUiSettings = [
 ];
 
 export interface SearchSourceDependencies extends FetchHandlers {
-  search: ISearchGeneric;
+  search: (request: IEsSearchRequest, options: ISearchOptions) => Promise<IEsSearchResponse>;
```

Comment:
@lukeelmers why did you do this change here?
`ISearchGeneric` is the exact equivalent of this more explicit typescript.

Reply:
It's not quite the exact equivalent -- `ISearchGeneric` returns an observable, however on the server, the low-level search returns a promise.

In order to ensure the same behavior of `SearchSource` on both client & server, I had to wrap the dependency into something that always returns a promise. 

That said, I did not include the ability to provide generic type params like `ISearchGeneric` does, and probably should have. I will open a PR to update this.

Reply:
(Also per our offline discussion, if low-level search on the server eventually moves to returning an observable as it does on the client, we can come back and adjust this)

Reply:
PR adding the generic type params: https://github.com/elastic/kibana/pull/79608